src/flamegraph/mod.rs

```diff
@@ -621,8 +577,35 @@ where
             }
 
             svg.write_event(&cache_a)?;
+        } else {
+            if let Event::Start(ref mut g) = cache_g {
+                // clear the BytesStart
+                g.clear_attributes();
+
+                if let Some(class) = frame_attributes.class {
+                    g.extend_attributes(std::iter::once(("class", class)));
+                }
+
+                // add optional attributes
+                if let Some(style) = frame_attributes.style {
+                    g.extend_attributes(std::iter::once(("style", style)));
+                }
+                if let Some(extra) = frame_attributes.g_extra {
+                    g.extend_attributes(extra.iter().map(|(k, v)| (k.as_str(), v.as_str())));
+                }
+            } else {
+                unreachable!("cache wrapper was of wrong type: {:?}", cache_g);
+            }
+
+            svg.write_event(&cache_g)?;
```

Comment:
All of the attributes that get added to the `<g>` element here need to also be added to the `<a>` for the `href` case. Now that `<a>` can take the place of the `<g>` as the container, it needs to have everything the `<g>` would have plus all of the stuff that only belongs on the `<a>`.

It might be helpful to pull this common "container" writing code out into a separate function so it can be reused in both cases.

```Rust
        // Use either <a> or <g> element for the container.
        if let Some(href) = frame_attributes.href {
            write_href_attributes(&mut cache_a, &frame_attributes, href);
            write_container_attributes(&mut cache_a, &frame_attributes, false);
            svg.write_event(&cache_a)?;
        } else {
            write_container_attributes(&mut cache_g, &frame_attributes, true);
            svg.write_event(&cache_g)?;
        }
```

```Rust
fn write_href_attributes<'a>(event: &mut Event, frame_attributes: &FrameAttributes, href: &str) {
    if let Event::Start(ref mut a) = event {
        a.clear_attributes();

        a.extend_attributes(args!(
            "xlink:href" => href,
            "target" => frame_attributes.target
        ));
        if let Some(extra) = frame_attributes.a_extra {
            a.extend_attributes(extra.iter().map(|(k, v)| (k.as_str(), v.as_str())));
        }
    } else {
        unreachable!("cache wrapper was of wrong type: {:?}", event);
    }
}

fn write_container_attributes<'a>(
    event: &mut Event,
    frame_attributes: &FrameAttributes,
    clear_attributes: bool,
) {
    if let Event::Start(ref mut c) = event {
        if clear_attributes {
            c.clear_attributes();
        }

        if let Some(id) = frame_attributes.id {
            c.extend_attributes(std::iter::once(("id", id)));
        }
        if let Some(class) = frame_attributes.class {
            c.extend_attributes(std::iter::once(("class", class)));
        }
        if let Some(extra) = frame_attributes.g_extra {
            c.extend_attributes(extra.iter().map(|(k, v)| (k.as_str(), v.as_str())));
        }
    } else {
        unreachable!("cache wrapper was of wrong type: {:?}", event);
    }
}
```